src/solver/conv_ck_igemm_fwd_bias_res_add_activ_fused.cpp

```diff
@@ -0,0 +1,481 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <vector>
+#include <cstdint>
+
+#include <miopen/check_numerics.hpp>
+#include <miopen/solver.hpp>
+#include <miopen/fusion/solvers.hpp>
+#include <miopen/generic_search.hpp>
+#include <miopen/conv/data_invoke_params.hpp>
+#include <miopen/solver/problem_description_interpreter.hpp>
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+#include <miopen/solver/ck_utility_common.hpp>
+#include <miopen/solver/implicitgemm_ck_util.hpp>
+#include <ck/tensor_operation/gpu/device/device_conv_fwd_bias_activation.hpp>
+#include <ck/library/tensor_operation_instance/gpu/grouped_convolution_forward_scaleadd_scaleadd_relu.hpp>
+#endif
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_BIAS_RES_ADD_ACTIV)
+
+namespace miopen {
+namespace solver {
+namespace fusion {
+
+using CK_OutLayout = ck::tensor_layout::convolution::NDHWGK;
+
+// DataType also applies to weights
+// AccumDataType also applies to added z & bias tensors
+template <typename DataType, typename AccumDataType = DataType>
+using DeviceOp = ck::tensor_operation::device::instance::DeviceOperationInstanceFactory<
+    ck::tensor_operation::device::DeviceGroupedConvFwdMultipleABD<
+        3,
+        ck::tensor_layout::convolution::NDHWGC,
+        ck::tensor_layout::convolution::GKZYXC,
+        ck::Tuple<CK_OutLayout, ck::tensor_layout::convolution::G_K>,
+        CK_OutLayout,
+        DataType,                                // in data type
+        DataType,                                // wei data type
+        ck::Tuple<AccumDataType, AccumDataType>, // z & bias tensors data type
+        DataType,                                // out data type
+        ck::tensor_operation::element_wise::PassThrough,
+        ck::tensor_operation::element_wise::PassThrough,
+        ck::tensor_operation::element_wise::
+            ScaleAddScaleAddRelu>>; // end DeviceOperationInstanceFactory
+
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+namespace {
+
+struct CKArgs
+{
+    CKArgs(const ProblemDescription& problem)
```

Comment:
@amberhassaan @JehandadKhan @atamazov weird how this get passed CI without 
either
`CKArgs(const miopen::conv::ProblemDescription& problem)`
or 
`using ProblemDescription = miopen::conv::ProblemDescription;`
at the top of file

Reply:
After merging https://github.com/ROCm/MIOpen/pull/2503, the code previously needed for compatibility with Fin was removed. It was intended that only Fin would use it, but it turned out that it was used in other places too